redash/handlers/access.py

```diff
@@ -0,0 +1,77 @@
+import logging
+from redash.handlers.base import BaseResource
+from redash.models import AccessPermission
+from flask import request
+from flask_restful import abort
+
+class AccessListResource(BaseResource):
+
+    def get(self, object_type, object_id):
+
+        permissions = AccessPermission.select(AccessPermission)\
+                .where(AccessPermission.object_id == object_id)\
+                .where(AccessPermission.object_type == object_type)\
+                .where(AccessPermission.grantor == self.current_user)
+
+        result = {}
+        for perm in permissions:
+            if perm.access_type not in result:
+                result[perm.access_type] = []
+            result[perm.access_type].append(perm.grantee.to_dict())
+
+        return result
+
+
+class AccessGrantResource(BaseResource):
+
+    def post(self, object_type, object_id):
```

Comment:
You need to make sure here that the user is admin or owner of the granted object.